### PR TITLE
feat(MapNode): added posibility so setup custom icons and load additional attributes for map-node

### PR DESCRIPTION
**Pull request by vrozaev**

### CODE REVIEW

File: packages/ui/src/ui/components/Icon/importGravityIcons.ts [hunk 1]
```diff
@@ -15,4 +15,5 @@ export const iconNames = {
     ['layout-side-content']: LayoutSideContent,
     ['sql']: AbbrSql,
     ['crowndiamond']: CrownDiamond,
+    ['python']: LogoPython,
```
Comment (ma-efremoff) [inline packages/ui/src/ui/components/Icon/importGravityIcons.ts:18]: It is never used directrly. Do we really need it?

File: packages/ui/src/ui/pages/navigation/content/MapNode/MapNodesTableIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import Icon from '../../../../components/Icon/Icon';
+import {getIconNameForType} from '../../../../utils/navigation/path-editor';
+
+const TRASH_PATH = '//tmp/trash';
+
+export function isTrashNode(item: any) {
```
Comment (ma-efremoff) [inline packages/ui/src/ui/pages/navigation/content/MapNode/MapNodesTableIcon.tsx:7]: It is always bad idea to use `any` this way. Required type might be easily inferred from implementation:
```ts
export function isTrashNode(item: {path: string}) {
    ...
}
```

File: packages/ui/src/ui/pages/navigation/content/MapNode/MapNodesTableIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    return item.path === TRASH_PATH;
+}
+
+export function isLinkToTrashNode(item: any) {
```
Comment (ma-efremoff) [inline packages/ui/src/ui/pages/navigation/content/MapNode/MapNodesTableIcon.tsx:11]: `any`!

File: packages/ui/src/ui/pages/navigation/content/MapNode/MapNodesTableIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    return item.targetPath === TRASH_PATH;
+}
+
+export function renderMapNodesTableIcon(item: any) {
```
Comment (ma-efremoff) [inline packages/ui/src/ui/pages/navigation/content/MapNode/MapNodesTableIcon.tsx:15]: `any`!
Let's keep the method on its previous place. And add call of corresponding `UIFacotry`-method, like:
```ts
export function renderMapNodesTableIcon(item) {
    let icon = UIFactory.getNavigationMapNodeSettings()?.renderNodeIcon(item);
    if (icon) {
       // do nothing
    } else if ...
}
```

File: packages/ui/src/ui/components/Icon/importGravityIcons.ts (hunk 1 above)
Comment (vrozaev) [reply packages/ui/src/ui/components/Icon/importGravityIcons.ts:18]: You are right, it's not neccesery, I will roll it out.

File: packages/ui/src/ui/UIFactory.tsx [hunk 5]
```diff
@@ -426,6 +427,11 @@ export interface UIFactory {
         menuItems: Array<DropdownMenuItem>;
         renderModals: () => React.ReactNode;
     };
+
+    getMapNodeSettings(): {
+        additionalAttributes: unknown[];
+        renderMapNodesTableIcon: (item: unknown) => React.ReactNode;
```
[the remark this answers is not in the document]
Comment (vrozaev) [reply packages/ui/src/ui/UIFactory.tsx:433]: I will move this file packages/ui/src/ui/utils/navigation/content/map-nodes/node.js to typescript, that will give us accurate types.

**Review by ma-efremoff — APPROVED** [review]

**Pull request merged**